integration/oasis/stationgraph/node.go

```diff
@@ -35,23 +35,25 @@ func newNode() *node {
 		chargeInfo: chargeInfo{
 			arrivalEnergy: 0.0,
 			chargeTime:    0.0,
-			chargeEnergy:  0.0,
+			targetState: chargingstrategy.State{
+				Energy: 0.0,
+			},
 		},
 	}
 }
 
 // Function isLocationReachable is used to test whether target node is reachable
 func (n *node) isLocationReachable(distance float64) bool {
-	return n.chargeEnergy > distance
+	return n.targetState.Energy > distance
```

Comment:
`Energy` is in meter?

Reply:
Good suggestion.  Expose internal state directly is not a good idea.  Record which in https://github.com/Telenav/osrm-backend/issues/211.

Will provide function like `MaxDriveDistance` for struct `State`